pkg/apis/messaging/v1beta1/kafka_channel_lifecycle.go

```diff
@@ -99,8 +99,18 @@ func (kcs *KafkaChannelStatus) GetCondition(t apis.ConditionType) *apis.Conditio
 }
 
 // IsReady returns true if the resource is ready overall.
-func (kcs *KafkaChannelStatus) IsReady() bool {
-	return kcs.GetConditionSet().Manage(kcs).IsHappy()
+func (kc *KafkaChannel) IsReady() bool {
+	kcs := kc.Status
+	return kcs.ObservedGeneration == kc.Generation &&
+		kcs.GetCondition(KafkaChannelConditionReady).IsTrue()
+}
+
+// IsFailed returns true if the resource has observed the latest generation
+// and ready is false.
+func (kc *KafkaChannel) IsFailed() bool {
+	kcs := kc.Status
+	return kcs.ObservedGeneration == kc.Generation &&
+		kcs.GetCondition(KafkaChannelConditionReady).IsFalse()
```

Comment:
```suggestion
		return !kcs.GetConditionSet().Manage(kcs).IsHappy()
```
as mentioned above

Reply:
Not sure why the API was actually called `Happy`... :thinking:  But in the fashion of `isFailed` we might add a `IsUnhappy` to upstream? :thinking: